From 2c55d7696dc1a3ef75e728eecf90cb9a99f41d29 Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Sun, 11 Oct 2015 21:59:57 +0800
Subject: [PATCH 65/80] gpio: zynq: make sure init_gpio done before USB and
 after pinctrl

If init_gpio is called before init_pinctrl, probing will be delayed,
since init_gpio depends on init_pinctrl. So this patch is to put
GPIO_init in the right place by module_init, and then GPIO will
be ready before USB_init being called.

Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/gpio/gpio-zynq.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpio/gpio-zynq.c b/drivers/gpio/gpio-zynq.c
index 184c4b1..f3dd0db 100644
--- a/drivers/gpio/gpio-zynq.c
+++ b/drivers/gpio/gpio-zynq.c
@@ -720,7 +720,7 @@ static int __init zynq_gpio_init(void)
 {
 	return platform_driver_register(&zynq_gpio_driver);
 }
-postcore_initcall(zynq_gpio_init);
+module_init(zynq_gpio_init);
 
 MODULE_AUTHOR("Xilinx Inc.");
 MODULE_DESCRIPTION("Zynq GPIO driver");
-- 
2.0.2

